subsys/fs/fcb/fcb_elem_info.c

```diff
@@ -1,58 +1,43 @@
 /*
- * Copyright (c) 2017 Nordic Semiconductor ASA
+ * Copyright (c) 2017-2023 Nordic Semiconductor ASA
  * Copyright (c) 2015 Runtime Inc
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
+#include <stdbool.h>
 #include <zephyr/sys/crc.h>
 
 #include <zephyr/fs/fcb.h>
 #include "fcb_priv.h"
 
+#define FCB_FIXED_ENDMARKER 0xab
+
 /*
- * Given offset in flash sector, fill in rest of the fcb_entry, and crc8 over
- * the data.
+ * Given the offset in flash sector, crc8 over the data.
  */
-int
-fcb_elem_crc8(struct fcb *fcb, struct fcb_entry *loc, uint8_t *c8p)
+static int
+fcb_elem_crc8(struct fcb *_fcb, struct fcb_entry *loc, uint8_t *tmp_str, size_t tmp_str_len,
```

Comment:
IMO passing the `tmp_str` is a little too much trying to get the code size equal. The function should be calculate the crc8 given a "correct" `fcb_entry`. It is cleaner to define a `len_str` when checking the `fcb_entry` and then a `buf[FCB_TMP_BUF_SZ]` when calculating the crc8.